### PR TITLE
feat(frontend): Wait for events before rendering messages

### PR DESCRIPTION
**Pull request by amanape**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**
I created a new hook to manage the rate at which the messages are received. If they are under the given threshold (e.g., 500 ms) then the state is loading. Otherwise, if they are above OR the time passed since the last message is > 500 ms, it is not loading

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:6abe70d-nikolaik   --name openhands-app-6abe70d   docker.all-hands.dev/all-hands-ai/openhands:6abe70d
```

### CODE REVIEW

File: frontend/__tests__/hooks/use-rate.test.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    act(() => {
+      result.current.record(9);
+    });
+
+    expect(result.current.items).toHaveLength(2);
+    expect(result.current.rate).toBe(5);
+    expect(result.current.lastUpdated).toBe(1000);
+  });
+
+  it("should update isExceeding after [threshold]ms of no activity", () => {
```
Comment (amanape) [inline frontend/__tests__/hooks/use-rate.test.ts:52]:
```suggestion
  it("should update isUnderThreshold after [threshold]ms of no activity", () => {
```

File: frontend/__tests__/hooks/use-rate.test.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    act(() => {
+      vi.advanceTimersToNextTimer();
+      vi.advanceTimersToNextTimer();
+    });
+
+    expect(result.current.isUnderThreshold).toBe(false);
+  });
+
+  it("should return an isExceeding boolean", () => {
```
Comment (amanape) [inline frontend/__tests__/hooks/use-rate.test.ts:65]:
```suggestion
  it("should return an isUnderThreshold boolean", () => {
```

File: frontend/__tests__/hooks/use-rate.test.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  it("should update isExceeding after [threshold]ms of no activity", () => {
+    const { result } = renderHook(() => useRate({ threshold: 500 }));
+
+    expect(result.current.isUnderThreshold).toBe(true);
+
+    act(() => {
+      vi.advanceTimersToNextTimer();
+      vi.advanceTimersToNextTimer();
+    });
```
Comment (amanape) [inline frontend/__tests__/hooks/use-rate.test.ts:60]:
```suggestion
    act(() => {
       // not sure if fake timers is buggy with intervals,
       // but I need to call it twice to register
      vi.advanceTimersToNextTimer();
      vi.advanceTimersToNextTimer();
    });
```

**Review by tofarr — APPROVED** [review]
🍰

**Pull request merged**